### ISSUE TITLE
toHexString.call not supported in latest driver

### ISSUE THREAD

**Issue by imlucas** [mongodb-js/extended-json#56]
via #31 

```
Can I ask why toHexString.call(v) instead of v.toHexString?

toHexString.call(v) is breaking for values of v minted with a later version of the ObjectId class (coming from the MongoDB driver). More recent versions of ObjectId do are not compatible with the toHexString from the version of ObjectId in the version of bson included in this package. Namely:

TypeError: this.id.charCodeAt is not a function
at ObjectID.toHexString (node_modules/bson/lib/bson/objectid.js:74:35)

this.id is now a Buffer and not a string.

Calling v.toHexString will ensure that the version of the toHexString method being called is compatible with the internal data structure of v.
```

**Comment by rueckstiess** [comment]
Running into the same problem in Compass now. Submitting PR...

**Comment by imlucas** [comment]
Fixed by #58
